test-butler-app-physical-devices/src/main/java/com/linkedin/android/testbutler/shell/ActivityManagerWrapper.java

```diff
@@ -38,11 +43,20 @@ private ActivityManagerWrapper(Method broadcastIntent, Object iActivityManager)
         this.iActivityManager = iActivityManager;
     }
 
-    int broadcastIntent(Object iApplicationThreadCaller, Intent intent,
-                        String resolvedType, Object iIntentReceiverResultTo, int resultCode,
-                        String resultData, Bundle map, String[] requiredPermissions,
-                        int appOp, Bundle options, boolean serialized, boolean sticky,
-                        int userId) throws RemoteException {
+    void broadcastIntent(Intent intent) throws RemoteException {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            broadcastIntentPreM(null, intent, intent.getType(), null, 0, null, null, null, -1, true, false, CURRENT_USER_ID);
+        } else {
+            broadcastIntent(null, intent, intent.getType(), null, 0, null, null, null, -1, null, true, false, CURRENT_USER_ID);
+        }
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    private int broadcastIntent(Object iApplicationThreadCaller, Intent intent,
+                                String resolvedType, Object iIntentReceiverResultTo, int resultCode,
+                                String resultData, Bundle map, String[] requiredPermissions,
+                                int appOp, Bundle options, boolean serialized, boolean sticky,
+                                int userId) throws RemoteException {
```

Comment:
since this is all internal anyway, would it be cleaner to remove all theese unused params from the broadcastIntent() methoids, and just put thee nulls directly inside of the call? that way you can keep these methods and params cleaer on the parts that youre actually interested in

Reply:
Sure. I was trying to keep those methods in sync with the actual IDL for that API, but it's just more confusing I guess.